File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(std::vector<int>& prices) {
        int mini=prices[0];
        int profit=0;
        for(int i=1;i<prices.size();i++){
            int diff=prices[i] - mini;
            profit= std::max(profit , diff);
            mini = std::min(mini, prices[i]);
        }
        return profit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int buyPrices = Integer.MAX_VALUE;
        int maxprofit =0; 
        for(int i = 0 ; i< prices.length; i++){
            if(buyPrices < prices[i]){
              int profit= prices[i] - buyPrices ;
              maxprofit = Math.max(maxprofit , profit);
            }
            else{
                buyPrices = prices[i];
            }
        }
        return maxprofit;
    }
}